script.js

```diff
@@ -1,4 +1,162 @@
+// Global Variables
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayersScore = [];
+
+// Helper Function
+var rollDice = function () {
+  console.log("Control flow: start of rollDice()");
+  // Random decimal between 0 and 6
+  var randomDecimal = Math.random() * 6;
+  // Random integer from 1 to 6
+  var randomInteger = Math.floor(randomDecimal) + 1;
+
+  console.log("rollDice output, randomInteger: ", randomInteger);
+  return randomInteger;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control flow: start of rollDiceForPlayer()");
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
```

Comment:
in this instance it might be better to use a for loop as opposed to a while loop. You already know how many times the loop will run!